connect/manifest.ts

```diff
@@ -0,0 +1,46 @@
+import axios from 'axios';
+
+export interface WebManifest {
+  icons?: { src: string; sizes?: string }[];
+}
+
+// Fetch the web manifest
+async function getManifestFile(url: string) {
+  try {
+    const response = await axios.get<WebManifest>(`${url}/manifest.json`);
+    return response.data;
+  } catch (err) {
+    const response = await axios.get<WebManifest>(`${url}/manifest.webmanifest`);
+    if (response.data) {
+      return response.data;
+    }
+  }
+}
+
+export async function getAppIconFromWebManifest(url: string): Promise<string> {
+  // Validate URL format
+  if (!/^https?:\/\/.*/.test(url)) {
+    throw new Error('Invalid URL format');
+  }
+  const UrlParsed = url.split('/');
+  const baseURL = `${UrlParsed[0]}${UrlParsed[1]}${UrlParsed[2]}`;
+  const manifest = await getManifestFile(baseURL);
```

Comment:
you could probably use the built-in URL class here. something like:

if (!URL.canPase(url) {
  throw new Error();
}
const {origin} = new URL(url);
const manifest = await getManifestFile(origin);

Reply:
![image](https://github.com/secretkeylabs/xverse-core/assets/6109710/27b57e92-e421-4d3f-81b2-f66523b5de01)